pyproject.toml

```diff
@@ -33,18 +33,18 @@ dependencies = [
     "numpy>=1.24",
     "pandas>=2.0",
     "xarray>=2023.04",
-    "netCDF4",
-    "packaging",
+    "netCDF4>=1.7",
+    "packaging>=22.0",
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
 all = [
-    "contextily",
-    "geopandas",
-    "IPython",  # 'ipython' is not the correct module name.
-    "pyarrow",
-    "rioxarray",
+    "contextily>=1.2",
```

Comment:
Need to pin to `contextily>=1.2`, otherwise this code block won't work as expected (there is a chance that an ImportError will be raised if `contextily=1.1` is installed, but xyzservices is not installed, breaking the _HAS_CONTEXTILY logic):

https://github.com/GenericMappingTools/pygmt/blob/97185e8437fe73cc8661e89a3f655ea2b51f25f4/pygmt/datasets/tile_map.py#L12-L19

Xref https://github.com/geopandas/contextily/pull/183, where `xyzservices` was included as a requirement of `contextily` (in [v1.2.0](https://github.com/geopandas/contextily/releases/tag/v1.2.0))

Reply:
Good to know that we require contexily>=1.2